pkg/urbit/noun/allocate.c

```diff
@@ -1492,6 +1492,103 @@ u3a_use(u3_noun som)
   }
 }
 
+/* _ca_wed_who(): unify [a] and [b] on [rod_u], keeping the senior
+**
+** NB: this leaks a reference when it unifies in a senior road
+*/
+static c3_o
+_ca_wed_who(u3a_road* rod_u, u3_noun* a, u3_noun* b)
+{
+  c3_t asr_t = ( c3y == u3a_is_senior(rod_u, *a) );
+  c3_t bsr_t = ( c3y == u3a_is_senior(rod_u, *b) );
+  c3_t nor_t = ( c3y == u3a_is_north(rod_u) );
+  c3_t own_t = ( rod_u == u3R );
+
+  //  both are on [rod_u]; gain a reference to whichever we keep
+  //
+  if ( !asr_t && !bsr_t ) {
+    //  keep [a]; it's deeper in the heap
+    //
+    //    (N && >) || (S && <)
```

Comment:
This comment is more like what I was expecting the code to do.  Doesn't heap depth change direction based on whether the road is north or south?

Reply:
The condition on the next line is `( (*a > *b) == nor_t )`, ie `( ((*a > *b) && nor_t) || (!(*a > *b) && !nor_t) )`.

Reply:
Oh, right.  I misread that.